### PR TITLE
Fix compiler warnings and make compilation work in travis (OS X)

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @vpodzime can review this?

**Comment by olehermanse** [comment]
@cf-bottom jenkins!

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1148/)

https://ci.cfengine.com/job/pr-pipeline/1148/

File: travis-scripts/script.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ export DIST_TARBALL
 
 if [ "$JOB_TYPE" = compile_only ]
 then
-    make CFLAGS=-Werror
+    make CFLAGS="-Werror -Wno-pointer-sign" -k
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply travis-scripts/script.sh]: No `-k` is an argument to make. (Keep going after 1 error to show all errors).

[the remark this answers is not in the document]
Comment (olehermanse) [reply travis-scripts/script.sh]: Fixed!

**Pull request merged**